x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/LocalExecutionPlanner.java

```diff
@@ -332,10 +333,28 @@ private PhysicalOperation planTopN(TopNExec topNExec, LocalExecutionPlannerConte
                 throw new UnsupportedOperationException();
             }
 
+            TopNEncoder encoder = switch (a.dataType().typeName()) {
+                case "ip": {
+                    yield TopNOperator.BYTESREF_FIXED_LENGTH_ENCODER;
+                }
+                case "text", "keyword": {
+                    yield TopNOperator.BYTESREF_UTF8_ENCODER;
+                }
+                case "version": {
```

Comment:
@luigidellaquila @bpintea `version` data type values will not be changed in any way.